fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/ConditionJsonProvider.java

```diff
@@ -43,7 +43,7 @@ static void write(JsonObject conditionalObject, ConditionJsonProvider @Nullable.
 		JsonArray array = new JsonArray();
 
 		for (ConditionJsonProvider condition : conditions) {
-			array.add(condition.toJson());
+			if (condition != null) array.add(condition.toJson());
```

Comment:
Hmmm? No condition is supposed to be null, it's the vararg that is nullable (otherwise it would have been `@Nullable ConditionJsonProvider...`)

Reply:
Ah, makes sense. Changing this to requireNonNull.

Reply:
If you want - otherwise the crash is pretty straightforward imo. :smile: